### ISSUE TITLE
FileProvider writeBuffer method fails in VS Code

### ISSUE THREAD

**Issue by fredrikhr** [AnWeber/vscode-httpyac#98]
When I attempt to write a buffer to disk using the `fileProvider.writeBuffer` function from within VS Code, the function fails with
```
Error - No valid uri
Error: No valid uri
	at c:\Users\<Username>\.vscode\extensions\anweber.vscode-httpyac-4.9.0\dist\extension.js:275:7208
	at Generator.next (<anonymous>)
	at s (c:\Users\Fredrik\.vscode\extensions\anweber.vscode-httpyac-4.9.0\dist\extension.js:1:1393)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

The argument I pass into the `writeBuffer` is an Httpyac `PathLike` constructed from `fileProvider.joinPath`.

However, when I run the command from the commandline using the Httpyac CLI everything works fine.

**Comment by AnWeber** [comment]
What type is the passed PathLike? The error is thrown if the type is neither string nor a [vscode.URI](https://github.com/AnWeber/vscode-httpyac/blob/main/src/io/fileProvider.ts#L109-L120). I will try to recreate it locally.

**Comment by fredrikhr** [comment]
``` js
const filePath = fileProvider.joinPath(config.envDirName, '.env');
await fileProvider.writeBuffer(filePath, myBuffer);
```

**Comment by fredrikhr** [comment]
I also attempted to do 
```
const filePath = fileProvider.joinPath(config.envDirName, '.env');
await fileProvider.writeBuffer(fileProvider.fsPath(filePath), myBuffer);
```

**Comment by fredrikhr** [comment]
But finally, this works:
``` js
const fs = require('fs/promises');
await fs.writeFile(fileProvider.fsPath(filePath), myBuffer);
```
But that goes against the philosophy of abstracting away the filesystem in the first place, so currently I do the three parts here in nested try-catches (i.e. falling back to node's fs module if all else fails).

**Comment by AnWeber** [comment]
When calling `fileProvider.joinPath(config.envDirName, '.env')` the method `vscode.URI.file('env') is called
![image](https://user-images.githubusercontent.com/13434739/148035611-ca18b3b3-5360-42dc-a219-b2cdee640f56.png)

VSCode, unlike `fs`, does not take the execution path into account. But I think that even with `fs` the desired directory is not written, because the execution path for me is `%localappdata%\Programs\Microsoft VS Code`. Using httpyac cli the execution path should be the desired folder.



Here's what would work.
```
 const filePath = fileProvider.joinPath(httpFile.rootDir, config.envDirName, '.env');
  fileProvider.writeBuffer(filePath , ...)
```

**Comment by fredrikhr** [comment]
I use `utils.toAbsoluteFilename` instead. However, the resulting `PathLike` is still not accepted by VS Code.

``` js
/**
 * @param {import('httpyac').PathLike} fileName
 * @param {import('httpyac').Variables} variables
 * @param {string[]} [prefix]
 */
async storeDotenvFile(fileName, variables, prefix) {
  const lines = toDotenvLines(variables);
  if (prefix) {
    lines.unshift(...prefix);
  }
  const text = lines.join(EOL) + (lines ? EOL : '');
  const buffer = Buffer.from(text, 'utf-8');
  const fileNameAbsolute = await utils.toAbsoluteFilename(fileName, httpFile.rootDir);
  if (!fileNameAbsolute) {
    log.warn('in storeDotenvFile: fileNameAbsolute is falsy');
    return;
  }
  // const fileFsPath = fileProvider.fsPath(fileNameAbsolute);
  await fileProvider.writeBuffer(fileNameAbsolute, buffer); // THIS DOES NOT WORK
}
```

I logged the values here's the log output in VS Code:
```
INFO: in storeDotenvFile: fileProvider.writeBuffer(fileNameAbsolute, buffer), value of fileNameAbsolute:
{
  "$mid": 1,
  "path": "/c:/some dir/test/.env/.env",
  "scheme": "file"
}
WARN: Error - No valid uri
Error: No valid uri
	at c:\Users\<username>\.vscode\extensions\anweber.vscode-httpyac-4.9.0\dist\extension.js:275:7208
	at Generator.next (<anonymous>)
	at s (c:\Users\<username>\.vscode\extensions\anweber.vscode-httpyac-4.9.0\dist\extension.js:1:1393)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
INFO: in storeDotenvFile: fileProvider.writeBuffer(fileFsPath, buffer), value of fileFsPath:
c:\some dir\test\.env\.env
```

So if I take the absolute `PathLike` and convert to a string using `fileProvider.fsPath` then it works.

**Comment by AnWeber** [comment]
Could you please give a concrete example for fileName? Is this a string or a vscode.URI. Is the code executed inside an http file or in the context of an httpyac plugin or .httpyac.js? The NodeJS context is slightly different.

**Comment by fredrikhr** [comment]
Ah, the code is running as a plugin. To be more precise, I have a httpyac.js file that requires my plugin which adds a response hook.
The error happens when the hook is triggered. The log output is the real concrete path pointing to the default dotenv file (.env) in the directory pointed to by envDirName (set to .env). The rootDir is C:\some dir\test\, hence the path C:\some dir\test\.env\.env
Meaning the utils, fileProvider and log values here are all coming from the api argument passed on to the plugin.

**Comment by AnWeber** [comment]
Sorry. I must have been blind. The error was only a forgotten else block. I was just able to recreate it using the ResponseHook. But why I didn't see that this morning, I don't understand.

**Comment by AnWeber** [comment]
should be fixed in v4.9.1
